api/workload/v2/stream.go

```diff
@@ -0,0 +1,132 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/proto/api/workload"
+	"go.uber.org/multierr"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// streamManager manages connection streams
+type streamManager struct {
+	// Chan is a channel of streams for fetching X509 SVIDs. It is updated whenever a new stream is created.
+	Chan           chan *managedStream
+	ctx            context.Context
+	logger         *logrus.Logger
+	addr           string
+	reconnectChan  chan struct{}
+	connectionChan chan bool
+}
+
+type managedStream struct {
+	workload.SpiffeWorkloadAPI_FetchX509SVIDClient
+
+	closer io.Closer
+}
+
+// Close closes the stream and the underlying connection.
+func (s *managedStream) Close() error {
+	return multierr.Combine(
```

Comment:
Consider not adding a new dependency for this single use case that can likely be accomplished with a fmt.Errorf()